### PR TITLE
Add default tokenUrl in authorizationCode and password flows

### PR DESCRIPTION
**Pull request by kapilraajP**
## Purpose
- Fix https://github.com/ballerina-platform/openapi-tools/issues/788

**Default tokenUrl in password flows and refreshUrl in authorizationCode flows** 
In the earlier implementation we only provide default tokenUrl in `http:OAuth2ClientCredentialsGrantConfig`. `tokenUrl` can be found in `http:OAuth2PasswordGrantConfig` and  `refreshUrl` can be found in `http:OAuth2RefreshTokenGrantConfig` too.
For increased usability providing the `tokenUrl` and `refreshUrl` by default would be better as follows,

```bal
public type OAuth2RefreshTokenGrantConfig record {|
    *http:OAuth2RefreshTokenGrantConfig;
    string refreshUrl = "https://login.salesforce.com/services/oauth2/token";
|};
```
```bal
public type OAuth2RefreshTokenGrantConfig record {|
    *http:OAuth2PasswordGrantConfig;
    string tokenUrl = "https://test.salesforce.com/services/oauth2/token";
|};
```
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: openapi-cli/src/main/java/io/ballerina/openapi/generators/client/BallerinaAuthConfigGenerator.java [hunk 1]
```diff
@@ -380,6 +392,124 @@ private List<Node> getClientCredsGrantConfigFields() {
...
+     *
+     * @return {@link List<Node>}
+     */
+    private List<Node> getPasswordGrantConfigFields() {
+        List<Node> recordFieldNodes = new ArrayList<>();
+        Token semicolonToken = createToken(SEMICOLON_TOKEN);
+        Token equalToken = createToken(EQUAL_TOKEN);
+
+        recordFieldNodes.add(createIncludedRecordParameterNode(createEmptyNodeList(),
+                createIdentifierToken(ASTERISK_TOKEN.stringValue()),
```
Comment (aneeshafedo) [inline openapi-cli/src/main/java/io/ballerina/openapi/generators/client/BallerinaAuthConfigGenerator.java:439]:
```suggestion
                createToken(ASTERISK_TOKEN),
```

File: openapi-cli/src/main/java/io/ballerina/openapi/generators/client/BallerinaAuthConfigGenerator.java [hunk 2]
```diff
@@ -380,6 +392,124 @@ private List<Node> getClientCredsGrantConfigFields() {
...
+     *
+     * @return {@link List<Node>}
+     */
+    private List<Node> getRefreshTokenGrantConfigFields() {
+        List<Node> recordFieldNodes = new ArrayList<>();
+        Token semicolonToken = createToken(SEMICOLON_TOKEN);
+        Token equalToken = createToken(EQUAL_TOKEN);
+
+        recordFieldNodes.add(createIncludedRecordParameterNode(createEmptyNodeList(),
+                createIdentifierToken(ASTERISK_TOKEN.stringValue()),
```
Comment (aneeshafedo) [inline openapi-cli/src/main/java/io/ballerina/openapi/generators/client/BallerinaAuthConfigGenerator.java:498]:
```suggestion
                createToken(ASTERISK_TOKEN),
```

**Review by aneeshafedo — APPROVED** [review]

File: openapi-cli/src/test/resources/generators/client/auth/scenarios/oauth2/oauth2_authorization_code_without_tokenurl.yaml [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  headers: {}
+  securitySchemes:
+    oAuth2AuthCode:
+      type: oauth2
+      description: For more information, see https://api.slack.com/docs/oauth
+      flows:
+        authorizationCode:
+          authorizationUrl: https://login.salesforce.com/services/oauth2/autherize
+          tokenUrl: ""
+          scopes: {}
```
Comment (lnash94) [inline openapi-cli/src/test/resources/generators/client/auth/scenarios/oauth2/oauth2_authorization_code_without_tokenurl.yaml:118]: Shall we add a new line there and other relevant places?

Comment (kapilraajP) [reply openapi-cli/src/test/resources/generators/client/auth/scenarios/oauth2/oauth2_authorization_code_without_tokenurl.yaml]: Added new lines at bottom

**Review by lnash94 — APPROVED** [review]
LGTM

**Pull request merged**